enhance GROMACS easyblock to improve control over PLUMED support (native vs. patching)

ostueker opened the pull request:
Since [GROMACS 2025][1] one can compile GROMACS with a PLUMED interface without patching it by setting CMake option [`-DGMX_USE_PLUMED=ON`][2]. 
Applying [PLUMED patches to GROMACS 2025][3] is still possible in order to get the latest PLUMED functionality.

This PR adds a fourth option `'patch'` to the extra-option `plumed`.

* `plumed = None` (default) auto detects whether PLUMED is available and do the same as `plumed = True`
* ~`plumed = True` sets `-DGMX_USE_PLUMED=ON` for GROMACS >= 2025 or try to patch GROMACS with PLUMED's patches for older (<=2024.x) versions (depends whether the loaded version has patches for that version of GROMACS~
* `plumed = 'patch'` will try to patch GROMACS with PLUMED's patches regardless of the GROMACS version
* `plumed = False` will disable PLUMED functionality, even if it is found (neither apply patches nor set `-DGMX_USE_PLUMED=OFF` for GROMACS >=2025

`<edit>`_improved logic is [here](https://github.com/easybuilders/easybuild-easyblocks/pull/3984#issuecomment-3553835074)_`</edit>`

I think that's the best of both worlds. It behaves like before for older GROMACS releases (before 2025) and takes the minimal-invasive route (avoiding to patch) when set to `None` ~or `True`~.

[1]: https://manual.gromacs.org/2025.2/release-notes/2025/major/features.html#a-feature-limited-version-of-the-plumed-interface-is-available
[2]: https://manual.gromacs.org/2025.2/install-guide/index.html#building-with-plumed-support
[3]: https://www.plumed.org/doc-v2.10/user-doc/html/gromacs-2025-0.html

ostueker committed:
EB_GROMACS: finer control over PLUMED support (internal or patching)

ostueker committed:
GROMACS: improve PLUMED patching logic

ostueker committed:
remove print statements added for debugging

ostueker committed:
GROMACS: add option plumed = 'native'

jhmeinke commented:
Thanks for the update! 

I agree, the tricky part is `plumed = None`. I'm worried that somebody who has given `PLUMED` as a dependency  before and now switches to a version of GROMACS >= 2025 will not get the PLUMED functionality they are expecting if the default is to go with the native PLUMED support. 

If I hadn't come across this thread, I would have been wondering why some of our users are complaining about their simulations with GROMACS and PLUMED not working as expected anymore.

I would use patching as the default behavior when `PLUMED` is given as a dependency for all versions of GROMACS that don't have full native support for PLUMED and use `plumed = 'native'` to opt in for now. Once native support is complete, it should of course be the default.

ostueker committed:
use native PLUMED by default with GROMACS 2026+

GROMACS 2025 was released on Feb 11, 2025. Changing EB's default behavior
of whether to apply patches or use GROMACS' native PLUMED support would
be surprising to users.

Changing the default with the yet unreleased GROMACS 2026 is safer.

In any case PLUMED patches for GROMACS 2025 and newer are based on
GROMACS' native support, which means that some PLUMED features that were
available with older GROMACS releases are not supported until GROMACS'
MDModule and the plugin mechansims have caught up again.

See also discussion in easybuilders/easybuild-easyblocks#3984

ostueker committed:
GROMACS: fix native PLUMED support instead of patching

ocaisa commented on easybuild/easyblocks/g/gromacs.py:
```diff
@@ -215,43 +217,81 @@ def configure_step(self):
+        # plumed = 'native' will enable GROMACS' native PLUMED support ('-DGMX_USE_PLUMED=ON')
+        # for GROMACS 2025 and newer. plumed = 'patch' specifically requests PLUMED patches.
+        # In auto-detect ('plumed = None' or not defined) will prefer native support for 2026
+        # and newer. Older versions of GROMACS patches are applied to enable PLUMED support.
```
```suggestion
        # and newer. For older versions of GROMACS, patches are applied to enable PLUMED support.
```

ocaisa reviewed (changes requested):
LGTM, just a trivial comment change, no need for further rebuilds

ocaisa committed:
Update easybuild/easyblocks/g/gromacs.py
